proxy/mux.go

```diff
@@ -141,13 +142,16 @@ func (mux *TCPMux) muxConnection(conn net.Conn) {
 	// make sure that we don't block indefinitely
 	conn.SetReadDeadline(time.Now().Add(time.Second * 5))
 
-	// Read in the mux header (a 6-byte representation of a TCP address)
-	muxHeader := make([]byte, 6)
-	if _, err := conn.Read(muxHeader); err != nil {
+	// Read in the authenticated mux header
+	authMuxHeader, err := auth.ReadMuxHeader(conn)
+	if err != nil {
 		log.WithError(err).Warn("Unable to read valid mux header. Closing connection")
 		conn.Close()
 		return
 	}
+	// TODO retrieve and validate the identity of the sender
```

Comment:
Yes, this will be accomplished in the authorization stage of this feature. The current PR is for the authentication framework; nothing is rejected yet.